### PR TITLE
add `cargo check --release` to ci

### LINKED ISSUE #1728: Add `release compile` to CI
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

Currently, datafusion's ci doesn't have a `release compile`, so it's very possible to introduce bugs such as #1723

**Describe the solution you'd like**

Add `cargo build --release` to https://github.com/apache/arrow-datafusion/blob/b9a8f151893b2487ee4d5e9018af71713565e26d/.github/workflows/rust.yml#L61

**Describe alternatives you've considered**

No

**Additional context**

No

### PR DESCRIPTION
**Pull request by xudong963**
# Which issue does this PR close?

Closes #1728 

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/src/physical_plan/tdigest/mod.rs [hunk 1]
```diff
@@ -266,7 +266,6 @@ impl TDigest {
     }
 
     fn merge_sorted_f64(&self, sorted_values: &[OrderedFloat<f64>]) -> TDigest {
-        #[cfg(debug_assertions)]
```
Comment (xudong963) [inline datafusion/src/physical_plan/tdigest/mod.rs:269]: Delete it temporarily to see if CI will report an error

Comment (xudong963) [reply datafusion/src/physical_plan/tdigest/mod.rs:269]: it works. 
image
cc @alamb

**Comment by Jimexist** [comment]
I don't think you need build per se - using check is good enough

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -58,12 +58,18 @@ jobs:
...
+      - name: Build workspace in debug mode
         run: |
           cargo build
         env:
           CARGO_HOME: "/github/home/.cargo"
-          CARGO_TARGET_DIR: "/github/home/target"
+          CARGO_TARGET_DIR: "/github/home/target/debug"
+      - name: Build workspace in release mode
+        run: |
+          cargo build --release
```
Comment (alamb) [inline .github/workflows/rust.yml:69]:
```suggestion
          cargo check --release
```

**Review by alamb — APPROVED** [review]
Thanks @xudong963

**Pull request merged**